modules/gs1-importer.js

```diff
@@ -146,6 +141,177 @@ function parseBatches(vocabularyElementList) {
     return batches;
 }
 
+/**
+ * Zero knowledge processing
+ * @param senderId
+ * @param event
+ * @param eventId
+ * @param categories
+ * @param importId
+ * @param globalR
+ * @param batchVertices
+ * @return {Promise<void>}
+ */
+async function zeroKnowledge(
```

Comment:
It would be great if we could modularize this one to make it more readable. It is a big method. Also, there is this repeating logic for inputQuantities. It would be awesome to extract it.

Reply:
We intentionally left it like that for now in order to test and see whether we covered all the use cases. Refactoring phase is planned for next week.